client/post-editor/editor-sidebar/constants.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+export const NESTED_SIDEBAR_NONE = 'NESTED_SIDEBAR_NONE';
```

Comment:
Would it make sense to change these to `X_HIDDEN` & `X_VISIBLE` or `X_EXPANDED` or something similar?
If in the future the hide/show logic changes to include more than the condition of just revisions/no revisions we might stumble in to weird cases. I could be completely wrong here, just offering food for thought :)